commit 8989e0c7eca187545d9867df22fe358ea880ea4e (refs/changes/20/4920/2)
Author: Marsell Kukuljevic <marsell@joyent.com>
Date:   2018-10-04T18:07:11+00:00 (1 year ago)
    
    OS-7278 'imgadm create' does not copy parent VM's bootrom into imgmanifest
    Reviewed by: Mike Gerdts <mike.gerdts@joyent.com>
    Approved by: Mike Gerdts <mike.gerdts@joyent.com>

diff --git a/src/img/lib/imgadm.js b/src/img/lib/imgadm.js
index aad2eebb..cc9e9fd5 100644
--- a/src/img/lib/imgadm.js
+++ b/src/img/lib/imgadm.js
@@ -3493,6 +3493,16 @@ IMGADM.prototype.createImage = function createImage(options, callback) {
                 log.debug({brand: m.requirements.brand},
                     'set image requirements.brand to source VM brand');
             }
+            if (vmInfo.brand === 'bhyve' && vmInfo.bootrom
+                && !(options.manifest.requirements
+                    && options.manifest.requirements.hasOwnProperty('bootrom')))
+            {
+                if (!m.requirements)
+                    m.requirements = {};
+                m.requirements.bootrom = vmInfo.bootrom;
+                log.debug({bootrom: m.requirements.bootrom},
+                    'set image requirements.bootrom to source VM bootrom');
+            }
             if (incremental) {
                 if (!originInfo) {
                     next(new errors.VmHasNoOriginError(vmUuid));
